content/en/blog/2024/cve-2024-36129/index.md

```diff
@@ -0,0 +1,127 @@
+
+Note that if you require authentication, an attacker would need to have valid
+credentials in order to exploit the vulnerability using the HTTP protocol. For
+gRPC, the exploitable code is executed before authentication.
+
+## What does it mean to me?
+
+If you manage a Collector that has an interface to the public internet, you
+should upgrade it as soon as feasible, and consider setting the parameter
+`max_request_body_size` on HTTP receivers, such as the OTLP receiver, to a value
+that makes sense to your workload. Up to v0.101.0, this setting applied only to
+the payload size sent by the client, which could often be compressed.
+
+Starting from v0.102.0, this setting applies to uncompressed, compressed, and
+decompressed payload sizes and we are establishing a default value of 20 MiB for
+this. This new default characterizes a breaking change, as clients sending
+payloads bigger than 20 MiB will start seeing an error. While we believe most
+authentic requests will be way within this limit, it’s still wise to monitor
+your Collector for increased error rates after this update. Here’s an example of
+a configuration setting a different limit to this field:
+
+```yaml
+receivers:
+  otlp:
+    protocols:
+      http:
+        endpoint: localhost:4318
+        max_request_body_size: 10485760 # 10 MiB
+```
+
+For gRPC receivers, it’s sufficient to upgrade to v0.102.1, as there’s a default
+value being applied to the message size already.
+
+If your Collector instances are receiving data only from trusted clients, like
+your own applications, you are still encouraged to upgrade to the latest
+Collector version but you can do it at your regular pace.
+
+If you are using a custom distribution and building it with the
+[OpenTelemetry Collector Builder (ocb)](/docs/collector/custom-collector/),
+you can add a
+[“replaces”](https://github.com/open-telemetry/opentelemetry-collector/tree/main/cmd/builder#configuration)
+entry pointing to the latest version of the
+[confighttp](http://go.opentelemetry.io/collector/config/confighttp) and
+[configgrpc](http://go.opentelemetry.io/collector/config/configgrpc) Go module.
+If your base Collector version is at v0.96.0 or higher, we do not expect any
+compatibility issues by just bumping to the latest version.
+
+## Lessons learned
+
+During this process, we found a couple of gaps in the telemetry for the
+Collector, as well as to the options we provide to Collector admins. Concretely,
+we noticed that we do not have a good way to verify what’s the distribution of
+request sizes received by the Collector, which would have been useful to
+determine whether the change would break clients for a given Collector. We also
+noticed that we don’t provide a way for admins to completely disable
+compression, which would be a good way to mitigate an attack without having to
+upgrade the Collector. We are working to fill those gaps over the next releases.
```

Comment:
If we have issues on Github for these already, these could perhaps be linked here.

Reply:
I don't think we don't yet